algorithms/SeatingChart/main.py

```diff
@@ -217,15 +219,67 @@ def generate_seating_charts(room_map_csv, room_allotment_csv, registered_student
                         start_point = start_point ^ 1
 
             for i in range(seated, student_count):
-                print(
-                    f"Could not assign a seat for {course.get_next_student()} in {room} for {course.code} - {course.ic_email}. Assign manually in the room.")
-
+                key_dict = room + "-" + course.time 
+                if key_dict not in left_out_students:
+                    left_out_students[key_dict] = {}
+                if course.code not in left_out_students[key_dict]:
+                    left_out_students[key_dict][course.code] = []
+                
+                left_out_students[key_dict][course.code].append(course.get_next_student())
+                not_alloted_students += 1
+                # print(
+                #     f"Could not assign a seat for {course.get_next_student()} in {room} {course.time} for {course.code} - {course.ic_email}. Assign manually in the room.")
+        left_out_students_copy  = copy.deepcopy(left_out_students)
         if course.allotment_index < len(course.students):
             print(
                 f"Seating Arrangement Discrepancy - {len(course.students) - course.allotment_index} students after {course.get_next_student()} for {course.code} - {course.title}")
 
+    left_out_students_count = 0
+    for time, courses in left_out_students.items():
+
+        room = time.split("-")[0]
+        for course_code in courses:
+
+            count = 0
+            course = course_list.find_by_code(course_code)
+            keys = get_matched_rooms(room_map, room)
+            for key in keys:
+
+                chart = final_solution[course.time][key]
```

Comment:
Since we are using `course.time` instead of `time.split("-")[1]`, consider removing the time parameter from the keys of `left_out_students`. If this is done, the other comments regarding making this key a tuple can be ignored.

Reply:
I think it can be useful in the error file.

Reply:
Yes. In that case, consider making this a tuple as per the previous comments.